front-end/src/components/analysis-page/IntentMenu.test.js

```diff
@@ -120,6 +120,128 @@ describe("IntentMenu", () => {
+            // Create a deep copy to prevent change of star attributes for future tests
+            props.intents = JSON.parse(JSON.stringify(sampleIntents));
+            props.intents[0].star = true;
+            props.intents[1].star = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            // A button for each of Intents 0-1 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+        });
+
+        it ("should correctly display 2 intents when two are flagged", () => {
+            // Create a deep copy to prevent change of flag attributes for future tests
+            props.intents = JSON.parse(JSON.stringify(sampleIntents));
+            props.intents[0].flag = true;
+            props.intents[1].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for each of Intents 0-1 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+        });
+
+        it ("should display only one intent when a single intent is both starred and flagged", () => {
+            // Create a deep copy to prevent change of attributes for future tests
+            props.intents = JSON.parse(JSON.stringify(sampleIntents));
+            props.intents[0].star = true;
+            props.intents[0].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for Intent 0 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 1 ? 1 : 0);
+            });
+        });
+
+        it ("should display two intents when two intents are both starred and flagged", () => {
+            // Create a deep copy to prevent change of attributes for future tests
+            props.intents = JSON.parse(JSON.stringify(sampleIntents));
+            props.intents[0].star = true;
+            props.intents[0].flag = true;
+            props.intents[1].star = true;
+            props.intents[1].flag = true;
+            renderComponent(props);
+            userEvent.click(screen.getByTestId("star-filter"));
+            userEvent.click(screen.getByTestId("flag-filter"));
+            // A button for each of Intents 0-1 should display
+            Object.keys(props.intents).forEach((key) => {
+                expect(screen.queryAllByLabelText(`${props.intents[key].question}`).length).toBe(key < 2 ? 1 : 0);
+            });
+        });
+    });
+
```

Comment:
After making the redux changes, I think there should also be tests that the filtered view and filter buttons are consistent with one another even after exiting and re-opening the intent menu when in a filtered view.